core/src/main/java/com/linecorp/armeria/common/util/AbstractUnwrappable.java

```diff
@@ -47,6 +47,22 @@ protected AbstractUnwrappable(T delegate) {
         return result != null ? result : delegate.as(type);
     }
 
+    /**
+     * Unwraps this {@link AbstractUnwrappable} recursively and returns the innermost object being
+     * decorated.
+     */
+    public Unwrappable unwrap() {
```

Comment:
- How about just unwrapping once, so that the caller has more control?
- Could we describe what will be returned when `this` is the innermost?

Reply:
Pulled up this method to `Unwrappable` and delegate() method is deprecated.